.github/workflows/push-mpc-to-ecr.yml

```diff
@@ -1,51 +1,70 @@
+# Builds and pushes MPC Signer to Amazon ECR. Relies on an OIDC identity provider to authenticate to AWS. Role used
+# must have permissions to publish to public ECR.
+
 name: Build & Push MPC Signer to Amazon ECR
 
 on:
-  push:
-    branches: [ "matt/feature/mpc-signer" ]
+  # Run if PR is merged to feature/mpc-signer or main branches
+  pull_request:
+    branches: [ "feature/mpc-signer", "main" ]
+    types: [ closed ]
+  # Run manually
+  workflow_dispatch:
 
 env:
+  # AWS Role with permissions to publish to Pulbic ECR
   AWS_ROLE_ARN: arn:aws:iam::961171757719:role/GithubActions-ECRPublicPowerUser
+  # AWS Region
   AWS_REGION: us-east-1
+  # ECR Repository name
   ECR_REPOSITORY: mpc-firefly-signer
+  # ECR Repository alias (for public ECR)
   ECR_REGISTRY_ALIAS: f8q0b6a6
 
 permissions:
-  id-token: write   # This is required for requesting the JWT
-  contents: read    # This is required for actions/checkout
+  # Requests JWT for GitHub Actions to authenticate to AWS
+  id-token: write
+  # Requests read access to the repository
+  contents: read
 
 jobs:
-  push-to-ecr:
-    name: Push to ECR
+  build-and-push-to-ecr:
+    # Run if manually triggered or if PR is merged
+    if: ${{ github.event_name == "workflow_dispatch" || github.event.pull_request.merged }}
+    name: Build & Push to ECR
     runs-on: ubuntu-latest
-    environment: dev
+    # Set to "prod" environment if running on main branch, otherwise "dev"
+    environment: ${{ github.ref == "refs/heads/main" && "prod" || "dev" }}
```

Comment:
The main branch here is only for the syncronisation with the original repo,
but I'm fine with that for now